src/test/java/io/AssignmentReaderTest.java

```diff
@@ -0,0 +1,191 @@
+          final int numPartitions,
+          final int batchSize)
+          throws IOException {
+
+    final Tuple2<Assignment<FrT>, Assignment<FrT>> expectPrimAux = expectPrimaryAuxiliary(one);
+    final Assignment<FrT> expectPrimary = expectPrimAux._1;
+    final Assignment<FrT> expectAuxiliary = expectPrimAux._2;
+    final ArrayList<FrT> expectFullArray = new ArrayList<FrT>(expectPrimary.elements());
+    expectFullArray.addAll(expectAuxiliary.elements());
+    final Assignment<FrT> expectFull = new Assignment<FrT>(expectFullArray);
+
+    // Read the distributed version of the assignment.
+    final Tuple2<Assignment<FrT>, JavaPairRDD<Long, FrT>> primFull =
+        assignmentReader.readPrimaryAuxiliaryRDD(
+            1, one, getSparkContext(), numPartitions, batchSize);
+    final Assignment<FrT> primary = primFull._1;
+    final JavaPairRDD<Long, FrT> full = primFull._2;
+
+    // Convert auxiliary to an Assignment object, and compare to the expected
+    // data.
+    final ArrayList<Tuple2<Long, FrT>> fullA = new ArrayList<Tuple2<Long, FrT>>(full.collect());
+    final Assignment<FrT> fullLocal =
+        new Assignment<FrT>(Utils.convertFromPairs(fullA, fullA.size()));
+
+    System.out.println(" expectPrimary: " + String.valueOf(expectPrimary.elements()));
+    System.out.println(" primary: " + String.valueOf(primary.elements()));
+    System.out.println(" expectFull: " + String.valueOf(expectFull.elements()));
+    System.out.println(" fullA: " + String.valueOf(fullA));
+    System.out.println(" fullLocal: " + String.valueOf(fullLocal.elements()));
+
+    assertEquals(expectPrimary, primary);
+    assertEquals(expectFull, fullLocal);
+  }
+
+  @Test
+  public void testAssignmentReaderBN254a() throws IOException {
+    final var in = openTestFile("assignment_alt-bn128.bin");
+    final var binReader = new BN254aBinaryReader(in);
+    testReaderAgainstData(
+        BN254aFr.ONE, new AssignmentReader<BN254aFr, BN254aG1, BN254aG2>(binReader));
+  }
+
+  @Test
+  public void testAssignmentReaderBLS12_377() throws IOException {
+    final var in = openTestFile("assignment_bls12-377.bin");
+    final var binReader = new BLS12_377BinaryReader(in);
+    testReaderAgainstData(
+        BLS12_377Fr.ONE, new AssignmentReader<BLS12_377Fr, BLS12_377G1, BLS12_377G2>(binReader));
+  }
+
+  @Test
+  public void testAssignmentReaderRDDBN254a_8_8() throws IOException {
+    final var in = openTestFile("assignment_alt-bn128.bin");
+    final var binReader = new BN254aBinaryReader(in);
+
+    testReaderAgainstDataRDD(
+        BN254aFr.ONE, new AssignmentReader<BN254aFr, BN254aG1, BN254aG2>(binReader), 8, 8);
+  }
+
+  @Test
+  public void testAssignmentReaderRDDBN254a_2_2() throws IOException {
+    final var in = openTestFile("assignment_alt-bn128.bin");
+    final var binReader = new BN254aBinaryReader(in);
+
+    testReaderAgainstDataRDD(
+        BN254aFr.ONE, new AssignmentReader<BN254aFr, BN254aG1, BN254aG2>(binReader), 2, 2);
```

Comment:
Do I understand correctly that you are reading just a subset of the assignment here? (i.e. execute `PairRDDAggregator` with only 2 batches and 2 partitions, which won't load the full assignment, right?)

Reply:
The full assignment should always be read.  `numPartitions` should only control the partitioning of the full assignment, and `batchSize` controls how many values are loaded per batch (on the assumption that the whole assignment may not fit in local memory, so much be loaded in fragments).

Reply:
Ah right, of course... Got it, thanks!